product-service/src/dal/product.service.js

```diff
@@ -0,0 +1,30 @@
+import { Client } from 'pg';
+
+import dbConfig from '../constants/db-config';
+
+export const addProduct = async product => {
+    const client = new Client(dbConfig);
+
+    try {
+        await client.connect();
+        await client.query('begin');
+
+        const { rows: [newProduct] } = await client.query(`
+            insert into product_list(title, genre, price)
+            values('${product.title}', '${product.genre}', ${+product.price})
```

Comment:
This is a bad and dangerous practice to use string concatenation with variables for DB queries. It makes possible SQL Injection: https://owasp.org/www-community/attacks/SQL_Injection https://www.w3schools.com/sql/sql_injection.asp